### PR TITLE
Allow users to record audio as wav files from the asset manager for Java Lab only

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
This adds a new NPM package, [recorder-js](https://www.npmjs.com/package/recorder-js) to specifically handle WAV recording, as our current solution only supports MP3, and abstracts away the multiple recorder implementations so `AudioRecorder.jsx` can use a unified interface. 

## Links

https://codedotorg.atlassian.net/browse/CSA-560

## Testing story

Unit + tested locally on Chrome, Firefox, Safari, and Edge (Windows)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/components/recorders.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  constructor() {
+    this.recorder = new vmsg.Recorder({wasmURL: '/shared/wasm/vmsg.wasm'});
+  }
+
+  init = async stream => {
+    await this.recorder.initAudio();
+    await this.recorder.initWorker();
+  };
+
+  startRecording = async () => {
```
Comment (jmkulwik) [inline apps/src/code-studio/components/recorders.js:14]: This file seems like a section that would benefit from some testing to future-proof against future maintenance. AudioRecorder makes strict assumptions that the recorder classes here are implemented as they are (i.e. both classes need to have startRecording, stopRecording, etc. stopRecording returns an audio blob etc.). Testing all of this functionality might not be possible/worthwhile, but some would be useful.

Comment (sanchitmalhotra126) [reply apps/src/code-studio/components/recorders.js:14]: Good call - I'll add tests to make sure this implementation stays consistent.

**Comment by jmkulwik** [comment]
One other note: sound often works differently on different browsers and OS's. Before merging, make sure to check on Safari/Mac and Edge/Windows - happy to help with the windows testing if needed.

**Comment by sanchitmalhotra126** [comment]
> One other note: sound often works differently on different browsers and OS's. Before merging, make sure to check on Safari/Mac and Edge/Windows - happy to help with the windows testing if needed.

Good call! I will check on Firefox and Safari with MacOS. I could use some help with testing on Edge/Windows - we can coordinate offline.

**Review by jmkulwik — APPROVED** [review]
Nice! Thanks for the added tests too!

**Comment by sanchitmalhotra126** [comment]
> > One other note: sound often works differently on different browsers and OS's. Before merging, make sure to check on Safari/Mac and Edge/Windows - happy to help with the windows testing if needed.
> 
> Good call! I will check on Firefox and Safari with MacOS. I could use some help with testing on Edge/Windows - we can coordinate offline.

Works successfully on Chrome, Firefox, Safari and Edge

**Pull request merged**